### DISCUSSION TITLE
Find tab from context menu and set active

### DISCUSSION THREAD

**Discussion by Balastrong** [microsoft/vscode-discussions#683] [Extension Development QnA] [answered]
Hello! I'm working on an extension which adds `workbench.action.closeEditorsToTheLeft` to the context menu when right-clicking on a tab.

It works ok, but starts closing tabs from the currently active one, not from the tab that got clicked.

I'd like to select the tab where I right-clicked first, and then run the command with 
```js
  await vscode.commands.executeCommand(
    "workbench.action.closeEditorsToTheLeft"
  );
```

However I'm not able to select the tab where I right-clicked in some cases.

---

Things I already tried, I see I get a `path` field when I call the command registered from `vscode.commands.registerCommand(...)`

I can use it for regular text files as I can match the `tab.input.path` from `vscode.window.tabGroups.all.flatMap((tabGroup) => tabGroup.tabs)` and get tab.input.uri that I pass to `vscode.window.showTextDocument(uri)`

However, if I click on an extension window I get on the `path` value from the context `'astro-build.astro-vscode/extension'` but that tab in the tabs array has `input` as `undefined` and I don't know how to set that tab as active.

**Comment by burkeholland** [comment]
@isidorn who is the right engineering contact to ping here. This looks to me like it _should_ work.

**Reply by isidorn** [reply]
@lramos15 should be able to help here since he owns the tabs api :)

**Reply by lramos15** [reply, marked as answer]
Unfortunately we don't support this case yet. We only support making active text documents or notebook documents. The tabs API as it stands is mostly readonly besides close, but we have a few issues tracking various things people want added, and maybe `makeActive()` or something of the likes is one of them. Feel free to open a feature request for it!

**Reply by marlindaleni68-png** [reply]
Puli kan
